### DISCUSSION TITLE
[feature_request] GraphQL fetch user language information

### DISCUSSION THREAD

**Discussion by rickstaa** [community/community#18230] [Apps, API and Webhooks]
There is currently no way to fetch the languages a specific user uses. The current `repositories/languages` object of the GraphQL API will return the languages of a repository. I think allowing users to filter only the languages used in commits made by their user account would improve the GraphQL API. It would provide users with more accurate Language proficiency information and improve the stats displayed by repositories like [github-readme-stats](https://github.com/anuraghazra/github-readme-stats). I'm unsure if this is technically possible given the current GraphQL design and performance requirements and the [linguist](https://github.com/github/linguist) tool but here is my proposal.

# Current behaviour

**Current graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    repositories(ownerAffiliations: OWNER, isFork: true, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

**Current GraphQL variables**

```graphql
{"login": "rickstaa"}
```

This will return the language stats per repository.

# New behaviour

Maybe an attribute like `isUser: true` could be added for fetching user-specific language information.

**New graphQL query:**

```graphql
query userInfo($login: String!) {
  user(login: $login) {
    repositories(ownerAffiliations: OWNER, isFork: true, first: 100) {
      nodes {
        name
        languages(first: 10, orderBy: {field: SIZE, direction: DESC}, isUser: true) {
          edges {
            size
            node {
              color
              name
            }
          }
        }
      }
    }
  }
}
```

**New GraphQL variables**

```graphql
{"login": "rickstaa"}
```

**Comment by Olf0** [comment]
I fully concur with the assessment and feature request: For many purposes the languages a *user* uses is of interest, not languages used in a *repository*.  This data is currently not accessible via the GraphQL API!

**Comment by StrangeRanger** [comment]
I agree that this should be a feature.

**Comment by jay23606** [comment]
Another nice feature would be an easy way to get the most recent commit date when inside Tree entries:

https://www.reddit.com/r/graphql/comments/1338vwr/new_to_graphql_and_trying_to_understand_github/

**Comment by Eminentzeal** [comment]
I buy this idea because I do a lot of work as a contributor to organizations repository, that shoul reflect on my profile as well

**Comment by Olf0** [comment]
@Eminentzeal, please look at issue https://github.com/anuraghazra/github-readme-stats/issues/1!

**Comment by Temerold** [comment]
👍 Agreed

**Reply by Temerold** [reply]
I am voicing my support in an additional way.

**Comment by thatmare** [comment]
I agree. I'm just learning to code and fork several projects to do so. After four projects and four months, the languages I used aren't in my top language stats. Any updates?

**Comment by spacey-sooty** [comment]
This would be super useful

**Comment by QwEekYhyo** [comment]
This should, indeed, be a feature.

**Comment by Destroy666x** [comment]
Forks are as important for GH as own repos, so definitely should be a thing.

**Reply by Eagle3386** [reply]
You do realize that your post is nothing but wasted, because voting works by clicking the 👍🏻 at the OP above, do you?!

**Reply by Destroy666x** [reply]
@Eagle3386 sorry, but you show blatant ridiculousness here, spamming the whole comment section with way more useless repeated comment. People should be able to comment whatever they want under a feedback discussion. While "+1" type of comments are not too useful, you commented mostly on a bunch that added some point(s). I have no idea what encouraged you to stop us in a hypocritical and comedic way. 

Also, do you realize:
- you can turn email notifications off
- you can unsubscribe from any discussions, knowing that 99% of replies in them will be support for the idea
?

**Comment by hokwanhung** [comment]
I cannot agree more; it indeed will reflect more precisely and could allow counting for forks and other behaviours, especially when the repository is company-based and users are just collaborators.

**Comment by BB-BenBridges** [comment]
Thumbs up to this. My current stats look like I can only write Python but that is far from the truth as I use a large variety of languages but in company owned orgs.

**Reply by Eagle3386** [reply]
You do realize that your post is nothing but wasted, because voting works by clicking the 👍🏻 at the OP above, do you?!

**Reply by BB-BenBridges** [reply]
Which I have done as well. Adding a further comment to show the different reasons this would be useful when this feature is developed could be useful to the devs. Certainly does not hurt.

**Reply by Eagle3386** [reply]
Nope, it _does_ hurt - the reasons are already clear, enough & unlike upvotes, new comments cause mails & notifications to anyone already subscribed due to comment, upvote, etc.

**Reply by BB-BenBridges** [reply]
"new comments cause mails" - like the 10 emails I just got from you? 😄 It would be one thing if there were multiple new comments on this per day, but it's been fairly steady at one a week. If the mods disagree, they can delete comments and pin one asking only to up-vote and not comment.

Not trying to be aggressive as I can see you were just trying to help. Just trying to show you the slight hypocrisy in your comment. The mods of this repo can manage the notifications if its a big issue for them. Same goes for anyone subbed to this issue.

**Comment by CodedRedGIT** [comment]
+1, this would be a super handy addition.

**Comment by mb6ockatf** [comment]
support this! 👍

**Comment by aamehedi** [comment]
Really need this :+1:

**Comment by InvalidJoker** [comment]
needed 👍

**Comment by Almighty-Shogun** [comment]
Really need this

**Comment by builtbybasit** [comment]
+1 We need this

**Comment by anahperes** [comment]
This idea would be useful!

**Comment by dsantosr** [comment]
+1

**Comment by dpakanati** [comment]
+1

**Comment by Dawsson** [comment]
would be very useful.

**Comment by BreeDurbin** [comment]
This would be super awesome. currently my contributions to organizations show up on my heat map but there is no way to display them in my profile with any other analytics tools which leads to skewed data which only represents my own repositories, but not the organizations I have contributed to.

Currently my displayed languages says its mostly c++ despite having made many contributions in kotlin and terraform for example.

**Comment by FlowerInTheRain** [comment]
I mostly work for organizations so it'd be a great improvement to my profile.

Huge +1

**Comment by Loomiyaa** [comment]
+1 would be a good feature

**Comment by GustavodCastro** [comment]
+1 I agree this would be a very helpful feature.

**Comment by W3ND31** [comment]
+1 would be an awesome feature

**Comment by tjklint** [comment]
I have something new to add to the table here. I'm a college students and nearly all my classes use GitHub classroom. When doing assignments and capstone project, they are all done on Forks. This is infuriating, not only do the commits not appear on my contribution graph, but when I try to show off the various languages I am using, it doesn't even show. Realistically, 50-70% of my languages used should C#, but due to the current implementation, it's a mere 5%.

**Comment by Zhou-Shilin** [comment]
+1 We need this feature.

**Comment by iwandejong** [comment]
+1

**Comment by GoodM4ven** [comment]
(loading Github...) "Please don't be another +1 guy" -MOTHER(smashes laptop)

**Comment by BEMZ01** [comment]
+1 This should be a feature

**Comment by kiril-buga** [comment]
This feature would be nice to have

**Comment by melarozz** [comment]
+1, would be really useful

**Comment by TheZoidMaster** [comment]
+1 this could be nice

**Comment by Hanamizu2002** [comment]
I strongly support this practice, and most of my contributions are in the organization.

**Comment by DevWannabe-dot** [comment]
+1, there is a wide range of possible uses to this addition. It is definitely worth considering by GitHub.

**Comment by edwloef** [comment]
+1 would be awesome!

**Comment by Tomfez** [comment]
+1 for this feature

**Comment by a19901201** [comment]
+1 I need this feature too.

**Comment by asvishnyakov** [comment]
As the one who mostly contribute into organization's / another user's repos, I would really appreciate to have this feature

**Comment by Erick-DAS** [comment]
+1

**Comment by ghanithan** [comment]
+1

**Comment by kboshold** [comment]
+1

**Comment by Pawel-Parma** [comment]
+1

**Comment by jizhuozhi** [comment]
+1

**Comment by jschwind** [comment]
+1, support this! 👍

**Comment by brendanlynn** [comment]
+1 coming from [anuraghazra/github-readme-stats](https://github.com/anuraghazra/github-readme-stats)! 👍

**Comment by Jaxkr** [comment]
+1

**Comment by lcsvoj** [comment]
+1

**Comment by Zarrock77** [comment]
+1

**Comment by vitalragaz** [comment]
+1

**Comment by joegar000** [comment]
+1

**Comment by fishshi** [comment]
+1

**Comment by IvanHCenalmor** [comment]
Support this! 👍 Indeed it would be super useful! 🤗

**Comment by freemanzMrojo** [comment]
+1

**Comment by Scoowy** [comment]
+1

**Comment by strachan** [comment]
+1

**Comment by ingenarel** [comment]
+1

**Comment by IvanildoBarauna** [comment]
+1

**Comment by Amronos** [comment]
+1

**Comment by grez-lucas** [comment]
+1

**Comment by alessandrobaldazzi** [comment]
+1

**Comment by eyalk007** [comment]
+1

**Comment by Vag-Soft** [comment]
+1

**Comment by fr3ddycucho** [comment]
+1

**Comment by Ryuuusuke** [comment]
+1

**Comment by AmeliaYeah** [comment]
+1

**Comment by TKanX** [comment]
+1

**Comment by 0foxes** [comment]
+1 this would be super helpful

**Comment by riseansmal** [comment]
+1

**Comment by plumj-am** [comment]
+1

**Comment by TheBooker66** [comment]
+1

**Comment by Rhernandez513** [comment]
+1

**Comment by SamDc73** [comment]
+1

**Comment by egerhether** [comment]
+1

**Comment by Adrwaan** [comment]
+1

**Comment by albert-byiringiro** [comment]
+1

**Comment by choikim0108** [comment]
+1

**Comment by bvngee** [comment]
+1! This would be great

**Comment by shelbeely** [comment]
+1

**Comment by udlose** [comment]
+1

**Comment by Garsooon** [comment]
+1

**Comment by yindaheng98** [comment]
+1

**Comment by RZachLamberty** [comment]
+1

**Comment by mcchampions** [comment]
I agree it.

**Comment by guischmid** [comment]
+1

**Comment by andregribeiro** [comment]
+1

**Comment by Trenza1ore** [comment]
+1

**Comment by Yuuzi261** [comment]
+1

**Comment by codelemon18** [comment]
+1

**Comment by Munkybooty** [comment]
+1

**Comment by gazer42** [comment]
+1

**Comment by jonathansilber** [comment]
+1

**Comment by UmiKami** [comment]
+1

**Comment by JustLikeCheese** [comment]
+1

**Comment by alt-0191** [comment]
+1

**Comment by filipopo** [comment]
+1

**Comment by sharktide** [comment]
+1

**Comment by it-me-joda** [comment]
+1

**Comment by mrinalTheCoder** [comment]
+1

**Comment by geniucker-dev** [comment]
+1

**Comment by onihilist** [comment]
+1

**Comment by nayrosk** [comment]
+1

**Comment by crafael23** [comment]
+1

**Comment by DereC4** [comment]
+1
